plumbing/format/packfile/patch_delta.go

```diff
@@ -334,28 +334,28 @@ func patchDeltaWriter(dst io.Writer, base io.ReaderAt, delta io.Reader,
 	// Avoid several interactions expanding the buffer, which can be quite
 	// inefficient on large deltas.
 	if b, ok := dst.(*bytes.Buffer); ok {
-		b.Grow(int(targetSz))
+		b.Grow(int(targetSz)) //nolint:gosec // G115: targetSz is delta target size
 	}
 
 	// If header still needs to be written, caller will provide
 	// a LazyObjectWriterHeader. This seems to be the case when
 	// dealing with thin-packs.
 	if writeHeader != nil {
-		err = writeHeader(typ, int64(targetSz))
+		err = writeHeader(typ, int64(targetSz)) //nolint:gosec // G115: targetSz is delta target size
 		if err != nil {
 			return 0, plumbing.ZeroHash, fmt.Errorf("could not lazy write header: %w", err)
 		}
 	}
 
 	remainingTargetSz := targetSz
 
-	hasher := plumbing.NewHasher(format.SHA1, typ, int64(targetSz))
+	hasher := plumbing.NewHasher(format.SHA1, typ, int64(targetSz)) //nolint:gosec // G115: targetSz fits in int64
```

Comment:
Similarly, `int64(targetSz)` in theory could overflow. I guess we could change `DecodeLEB128FromReader`'s return type to avoid the different casts around this package.